services/chat/api_send_messages.go

```diff
@@ -0,0 +1,159 @@
+package chat
+
+import (
+	"context"
+
+	"github.com/forPelevin/gomoji"
+
+	"github.com/status-im/status-go/eth-node/crypto"
+	"github.com/status-im/status-go/eth-node/types"
+	"github.com/status-im/status-go/protocol"
+	"github.com/status-im/status-go/protocol/common"
+	"github.com/status-im/status-go/protocol/protobuf"
+)
+
+type SendMessageResponse struct {
+	Chat     *Chat             `json:"chat"`
+	Messages []*common.Message `json:"messages"`
+}
+
+func (api *API) SendSticker(ctx context.Context, communityID types.HexBytes, chatID string, packID int32, hash string, responseTo string) (*SendMessageResponse, error) {
+	ensName, _ := api.s.accountsDB.GetPreferredUsername()
+
+	msg := &common.Message{
+		CommunityID: string(communityID.Bytes()),
+		ChatMessage: protobuf.ChatMessage{
+			ChatId:      chatID,
+			ContentType: protobuf.ChatMessage_STICKER,
+			Text:        "Update to latest version to see a nice sticker here!",
+			Payload: &protobuf.ChatMessage_Sticker{
+				Sticker: &protobuf.StickerMessage{
+					Hash: hash,
+					Pack: packID,
+				},
+			},
+			ResponseTo: responseTo,
+			EnsName:    ensName,
+		},
+	}
+
+	response, err := api.s.messenger.SendChatMessage(ctx, msg)
+	if err != nil {
+		return nil, err
+	}
+
+	return api.toSendMessageResponse(response)
+
+}
+
+func (api *API) toSendMessageResponse(response *protocol.MessengerResponse) (*SendMessageResponse, error) {
+	protocolChat := response.Chats()[0]
+
+	community, err := api.s.messenger.GetCommunityByID(types.HexBytes(protocolChat.CommunityID))
+	if err != nil {
+		return nil, err
+	}
+
+	pubKey := types.EncodeHex(crypto.FromECDSAPub(api.s.messenger.IdentityPublicKey()))
+	chat, err := api.toAPIChat(protocolChat, community, pubKey)
+	if err != nil {
+		return nil, err
+	}
+	return &SendMessageResponse{
+		Chat:     chat,
+		Messages: response.Messages(),
+	}, nil
+}
+
+func isTextOrEmoji(text string) protobuf.ChatMessage_ContentType {
+	contentType := protobuf.ChatMessage_TEXT_PLAIN
+	if gomoji.RemoveEmojis(text) == "" && len(gomoji.FindAll(text)) != 0 {
+		contentType = protobuf.ChatMessage_EMOJI
+	}
+
+	return contentType
+}
+
+func (api *API) SendMessage(ctx context.Context, communityID types.HexBytes, chatID string, text string, responseTo string) (*SendMessageResponse, error) {
+	ensName, _ := api.s.accountsDB.GetPreferredUsername()
+
+	msg := &common.Message{
```

Comment:
Similarly, I would probably move this into messenger for the most part, and just relay the arguments to messenger, but up to you